pkg/chunkenc/memchunk.go

```diff
@@ -140,34 +152,23 @@ func NewMemChunkSize(enc Encoding, blockSize, targetSize int) *MemChunk {
 		targetSize: targetSize, // Desired chunk size in compressed bytes
 		blocks:     []block{},
 
-		head: &headBlock{},
+		head:   &headBlock{},
+		format: chunkFormatV2,
 
 		encoding: enc,
-	}
-
-	switch enc {
-	case EncGZIP:
-		c.cPool = &Gzip
-	default:
-		panic("unknown encoding")
+		writers:  getWriterPool(enc),
+		readers:  getReaderPool(enc),
 	}
 
 	return c
 }
 
-// NewMemChunk returns a new in-mem chunk for query.
-func NewMemChunk(enc Encoding) *MemChunk {
-	return NewMemChunkSize(enc, 256*1024, 0)
-}
-
 // NewByteChunk returns a MemChunk on the passed bytes.
 func NewByteChunk(b []byte) (*MemChunk, error) {
 	bc := &MemChunk{
-		cPool:    &Gzip,
 		encoding: EncGZIP,
```

Comment:
I'm a little confused as to why this is set to EncGZIP here (maybe to default always to GZIP?) and I don't see anywhere that it gets reset if the encoding is different?

Reply:
yes this is leftover.

Reply:
Thanks !